util/ranger/detacher.go

```diff
@@ -298,10 +298,9 @@ func detachDNFCondAndBuildRangeForIndex(sctx sessionctx.Context, condition *expr
 }
 
 // DetachCondAndBuildRangeForIndex will detach the index filters from table filters.
-// If the top layer is DNF, we return a int slice which is eqAndInCount of every DNF item.
-// Otherwise just one number is returned.
+// If the top layer is CNF, we return a int of eqAndInCount, otherwise we return 0.
 func DetachCondAndBuildRangeForIndex(sctx sessionctx.Context, conditions []expression.Expression, cols []*expression.Column,
-	lengths []int) ([]*Range, []expression.Expression, []expression.Expression, int, error) {
+	lengths []int) (ranges []*Range, accesses, filters []expression.Expression, eqCount int, isDNF bool, err error) {
```

Comment:
This function returns 6 things. Could we merge the first five things into a structure?

Reply:
Done.